### DISCUSSION TITLE
Suggestion / Idea (connected to free brush auxiliary?)

### DISCUSSION THREAD

**Discussion by BlueFoxPrime** [f5/unovis#101] [Ideas]
Hey!

first of all congratulations for this awesome libary. I love it and can't wait to build more stuff with it!
You cover a lot of possible cases and offer flexible yet logic stucture. 

However, I have a suggestion, that (I think) would also help others in the future.

It would be awesome to be able to get the data of the actually displayed data pieces (e.g. displayed bars in a bar chart or visible part of line in a linechart) at onBrushEnd event (maybe other, because other libary component could change the data aswell)

Example for expected output after brush (screenshot below):

 {2016: [ streaming: {yTotal: number; yUpper: number, yLower: number}], 2017: [...], 2018: [...], 2019:[...]} 

![image](https://user-images.githubusercontent.com/112826784/212741861-0c1e6c21-6cd3-4daa-aa31-afae1f822f84.png)

The most important information for me is which categories (in this case "streaming") and which years are visible. 
In case I have overlooked how to achive this in a clean way, with the current possibilities please give me a hint.

Cheers,
Bluefox

**Comment by rokotyan** [comment]
Thanks @BlueFoxPrime!

Currently, you can get the selected X and Y ranges from the `onBrush` callback of `FreeBrush`, and then filter your data based on these ranges. Will that solve your problem? 

I don't think it'll be possible to add such information to the callbacks because components in Unovis don't communicate with each other, for example `FreeBrush` knows nothing about `StackedBar` and vise versa.

**Reply by BlueFoxPrime** [reply]
@rokotyan Thanks for you feedback!

Yes, that is possible! However my datasets can be very large, so it would be better if the displayed data could be exported rather than filtering it again (if the XY Container already filters it). If its not possible to access the displayed data from the XY Container (because the internal logic is not suitable for that), I will go with your suggestion.

**Reply by rokotyan** [reply]
@BlueFoxPrime I see, that makes perfect sense. Unfortunately I don't think we'll support that in the nearest future because architecturally it's difficult and Unovis was not designed to work with huge data sets (take a look at https://github.com/leeoniya/uPlot if you need high performance chart). But I'll keep your use case in mind for our future work, thanks!
